go/workflow.go

```diff
@@ -0,0 +1,270 @@
+		func() (GetStatusResponse, error) {
+			return queryGetStatus(ctx, customer)
+		})
+
+	// query handler for guest list
+	err = workflow.SetQueryHandler(ctx, common.QueryGetGuests,
+		func() ([]string, error) {
+			return queryGetGuests(ctx, customer)
+		})
+
+	// Block on everything. Continue-As-New on history length; size of activities in this workflow are small enough
+	// that we'll hit the length thresholds well before any size threshold.
+	for customer.AccountActive && info.GetCurrentHistoryLength() < common.EventsThreshold {
+		selector.Select(ctx)
+
+		if errSignal != nil {
+			logger.Error("Unrecoverable error in handling a signal.", "Error", errSignal)
+			return errSignal
+		}
+	}
+
+	// here because of events threshold, but account still active? Continue-As-New
+	if customer.AccountActive {
+		return workflow.NewContinueAsNewError(ctx, customer)
+	}
+
+	logger.Info("Loyalty workflow completed.", "Customer", customer)
+	return nil
+}
+
+func startGuestWorkflow(ctx workflow.Context, guest CustomerInfo) (err error, child workflow.ChildWorkflowFuture) {
+	childWorkflowOptions := workflow.ChildWorkflowOptions{
+		WorkflowID:            fmt.Sprintf(common.CustomerWorkflowIdFormat, guest.CustomerId),
+		ParentClosePolicy:     enums.PARENT_CLOSE_POLICY_ABANDON,
+		WorkflowIDReusePolicy: enums.WORKFLOW_ID_REUSE_POLICY_REJECT_DUPLICATE,
+	}
+	ctx = workflow.WithChildOptions(ctx, childWorkflowOptions)
+
+	child = workflow.ExecuteChildWorkflow(ctx, CustomerLoyaltyWorkflow, guest)
+	// wait for it to start before returning.
+	err = child.GetChildWorkflowExecution().Get(ctx, nil)
+	return err, child
+}
+
+func validateCustomerInfo(customer *CustomerInfo) {
+	customer.Status = status.NewStatus(customer.StatusLevel)
+	if len(customer.Guests) == 0 {
+		customer.Guests = make(map[string]struct{}, customer.Status.NumGuestsAllowed())
+	}
+}
+
+func signalAddPoints(ctx workflow.Context, c workflow.ReceiveChannel, customer *CustomerInfo) (err error) {
+	logger := workflow.GetLogger(ctx)
+	activities := Activities{}
+
+	var pointsToAdd int
+	c.Receive(ctx, &pointsToAdd)
+
+	logger.Info("Adding points to customer account.", "PointsAdded", pointsToAdd)
+	customer.LoyaltyPoints += pointsToAdd
+
+	statusChange := customer.Status.Update(customer.LoyaltyPoints)
+
+	if statusChange > 0 {
+		err = workflow.ExecuteActivity(ctx, activities.SendEmail,
+			fmt.Sprintf(common.EmailPromoted, customer.Status.Name())).
+			Get(ctx, nil)
+		if err != nil {
+			return errors.Wrap(err, "Error running SendEmail activity for status promotion.")
+		}
+	} else if statusChange < 0 {
+		err = workflow.ExecuteActivity(ctx, activities.SendEmail,
+			fmt.Sprintf(common.EmailDemoted, customer.Status.Name())).
+			Get(ctx, nil)
+		if err != nil {
+			return errors.Wrap(err, "Error running SendEmail activity for status demotion.")
+		}
+	}
+
+	return nil
+}
+
+func signalInviteGuest(ctx workflow.Context, c workflow.ReceiveChannel, customer *CustomerInfo) (err error) {
```

Comment:
```suggestion
func signalInviteGuest(ctx workflow.Context, c workflow.ReceiveChannel, customer *CustomerInfo) error {
```

There is no value in naming these returns